modules/core/test/test_mat.cpp

```diff
@@ -1719,17 +1719,23 @@ TEST(Mat, from_initializer_list)
 {
     Mat A({1.f, 2.f, 3.f});
     Mat_<float> B(3, 1); B << 1, 2, 3;
+    Mat_<float> C({3}, {1,2,3});
```

Comment:
Original issue reports about creation of 2D Mat. I believe we should test them too.